### ISSUE TITLE
[Bug]: MessageContainer in Exchange 2020 does not refer to the payload element in D2Payload

### ISSUE THREAD

**Issue by JosefKaltwasser** [DATEX-II-EU/DatexII#544]
### Bug description?

The MessageContainer in Exchange2020 declares its own 'payload' element instead of referring to the 'payload' element in DATEXII_3_D2Payload.xsd. But this element declaration contains all the constraints specified in the model that ensure unique id or id+version values in one message. So instead of declaring another payload element in the MessageContainer namespace, it should rather refer to the element declaration in the D2Payload namespace, i.e. instead of 
  <xs:complexType name="MessageContainer">
    <xs:sequence>
      <xs:element name="payload" type="com:PayloadPublication" minOccurs="0" maxOccurs="unbounded" />
it should better be
  <xs:complexType name="MessageContainer">
    <xs:sequence>
      <xs:element ref="d2:payload" minOccurs="0" maxOccurs="unbounded"/>
That just my initial thoughts. It steill bneeds to be tested.

### Version

3

**Comment by jonas-jaderberg** [comment]
Yes, we could. But, we haven't used ref to elements so far. And I don't think it's add any value. We use a mapping to an external class, PayloadPublication defined in common. PayloadPublication happens to be a rootElement, but we have nothing that says that we should use the rootElement in this case. We could, but I think we only do the mapping more complex without any added value.

**Comment by JosefKaltwasser** [comment]
Well. the added value is that we do not loose the uniqueness constraints, which Exchange2020 currently does. If these have a value, we shouldn't lose them. If they have no value, there should have been an issue to remove them from 16157-1, but that is not the case. If they have value, but we do not want to use 'ref', Exchange2020 would have to specify the constraints itself for its own payload element. That would be OK, but then the generation of the Exchange schema would need to be integrated into the tool (the choice of constraints depend on the payload model). Honestly speaking, this would be a better solution anyway. Currently, if you want to use Exchange2020, you have to create the payload schemas with the tool and then manufacture the corresponding Exchange2020 schemas yourself (different for Level A/B and Level C). On the German NAP we have produced the two sets of schemas and make them available for download, but this all offers quite a few potential sources for errors.

**Comment by jonas-jaderberg** [comment]
Yes, now I see the point.
I agree than that it would be better if we could use ref in this cases. I think we need to define a mapping and construct to be able to tell from the Exchange Model that we want to use an external class rootElement. One solution would be to introduce a new stereotype ExternalElement. In this case it should be named payload and be placed in the D2Payload ExternalNamespace. Mapping of an ExtyernalElement will then be done with ref. 
We do need to test this with XML databinding framweorks.
